packages/plugin_platform_interface/test/plugin_platform_interface_test.dart

```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// ignore_for_file: unreachable_from_main
```

Comment:
Why do we need this in a test file?

Reply:
Added a comment to explain this file-level ignore. I didn't want to add it to individual instances here because then it would show up in the docs since the unreachable code is in a docregion.

As to why this is needed for a test file: The lint is mostly targeting tests to catch instances where a test case is deleted or refactored, but its supporting infrastructure (e.g. public mock classes and helper methods) isn't removed.